sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/logicaltypes/NanosType.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.schemas.logicaltypes;
+
+import java.util.UUID;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.values.Row;
+
+/** Base class for types representing timestamps or durations as nanoseconds. */
+abstract class NanosType<T> implements Schema.LogicalType<T, Row> {
+  private final String identifier;
+
+  protected static final Schema SCHEMA;
+
+  static {
+    Schema schema = Schema.builder().addInt64Field("seconds").addInt32Field("nanos").build();
+    schema.setUUID(UUID.fromString("dc5915f1-6db8-3134-9444-4980fb088e92"));
+    SCHEMA = schema;
```

Comment:
this is a bit ugly - should we instead just make it an instance variable?

Reply:
Switched to instance variables. Proto conversion used by ByteBuddy still uses a static version of the schema for performance reasons.